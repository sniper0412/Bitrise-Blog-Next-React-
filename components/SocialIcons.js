import React from 'react';
import PropTypes from 'prop-types';
import SVG from 'react-inlinesvg';
import cx from 'classnames';

export default class SocialIcons extends React.Component {
  static propTypes = {
    vertical: PropTypes.bool,
    postTitle: PropTypes.string
  };

  state = {
    fixed: false
  };

  componentDidMount() {
    document.body.addEventListener('scroll', this.update);
  }

  componentWillUnmount() {
    document.body.removeEventListener('scroll', this.update);
  }

  update = () => {
    const fixed = !this.isVisible('.post-featured-image img') && !this.isVisible('.share-article');
    const hidden =
      this.isVisible('.share-article') ||
      this.isVisible('.footer') ||
      this.isVisible('.try-wrapper') ||
      this.isVisible('.disqus-container');

    this.setState({
      fixed,
      hidden
    });
  };

  isVisible = selector => {
    const navBarHeight = 128;
    const element = document.querySelector(selector);
    if (!element) return false;

    const { y, height } = element.getBoundingClientRect();

    return y + height >= navBarHeight && y <= window.innerHeight;
  };

  openPopup = url => {
    window.open(url, 'pop', 'width=600, height=400, scrollbars=no');
  };

  render() {
    const { vertical, postTitle = '' } = this.props;
    const { fixed, hidden } = this.state;

    return (
      <div className={cx(vertical ? 'share-article-vertical' : 'share-article')}>
        <div
          className={cx('social-icons', {
            fixed,
            hidden
          })}
        >
          <h3>{vertical ? 'Share' : 'Share this article'}</h3>
          <button
            className="facebookShareButton"
            onClick={() => this.openPopup(`https://www.facebook.com/sharer/sharer.php?u=${window.location}`)}
          >
            <div className="fb-container">
              <SVG src="/static/svg/social_icons/facebook.svg" />
            </div>
          </button>
          <button
            className="twitterShareButton"
            onClick={() => this.openPopup(`https://twitter.com/share?url=${window.location}&text=${postTitle}`)}
          >
            <div className="twitter-container">
              <SVG src="/static/svg/social_icons/twitter.svg" />
            </div>
          </button>
          <button
            className="linkedInShareButton"
            onClick={() =>
              this.openPopup(
                `https://www.linkedin.com/shareArticle?mini=true&url=${window.location}&title=${postTitle}`
              )
            }
          >
            <div className="linkedin-container">
              <SVG src="/static/svg/social_icons/linkedin.svg" />
            </div>
          </button>
          <button
            className="yCombinatorShareButton"
            onClick={() => this.openPopup('https://news.ycombinator.com/submit')}
          >
            <div className="yc-container">
              <SVG src="/static/svg/social_icons/ycombinator.svg" />
            </div>
          </button>
        </div>
      </div>
    );
  }
}
