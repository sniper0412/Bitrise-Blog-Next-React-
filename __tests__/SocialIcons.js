import renderer from 'react-test-renderer';

import SocialIcons from '../components/SocialIcons';

describe('SocialIcons component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<SocialIcons />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders vertically', () => {
    const tree = renderer.create(<SocialIcons vertical />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
