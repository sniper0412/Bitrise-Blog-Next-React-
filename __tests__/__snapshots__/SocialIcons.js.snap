// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialIcons component renders correctly 1`] = `
<div
  className="share-article"
>
  <div
    className="social-icons"
  >
    <h3>
      Share this article
    </h3>
    <button
      className="facebookShareButton"
      onClick={[Function]}
    >
      <div
        className="fb-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="twitterShareButton"
      onClick={[Function]}
    >
      <div
        className="twitter-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="linkedInShareButton"
      onClick={[Function]}
    >
      <div
        className="linkedin-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="yCombinatorShareButton"
      onClick={[Function]}
    >
      <div
        className="yc-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
  </div>
</div>
`;

exports[`SocialIcons component renders vertically 1`] = `
<div
  className="share-article-vertical"
>
  <div
    className="social-icons"
  >
    <h3>
      Share
    </h3>
    <button
      className="facebookShareButton"
      onClick={[Function]}
    >
      <div
        className="fb-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="twitterShareButton"
      onClick={[Function]}
    >
      <div
        className="twitter-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="linkedInShareButton"
      onClick={[Function]}
    >
      <div
        className="linkedin-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
    <button
      className="yCombinatorShareButton"
      onClick={[Function]}
    >
      <div
        className="yc-container"
      >
        <span
          className="isvg loading"
        />
      </div>
    </button>
  </div>
</div>
`;
